### DISCUSSION TITLE
Where are the new allowBuild() checks?

### DISCUSSION THREAD

**Discussion by LEOcab** [GriefPrevention/GriefPrevention#1834] [Q&A]
I'm trying to update my fork and I see you guys completely changed the Claim::allowBuild() method. The checks used to be inside that method, now it calls checkPermission(), which in turn creates a ClaimPermissionCheckEvent but not only does that class not exist in the 'events' folder, but Github search finds 0 references to it... I don't know what I'm doing wrong but I need to find out where that class is.

Can somebody help me out here, please?

**Comment by Jikoo** [comment]
[Right in with all the other events](https://github.com/TechFortress/GriefPrevention/blob/6b87e036fa7c747c92be025e9266d98adf099b0c/src/main/java/me/ryanhamshire/GriefPrevention/events/ClaimPermissionCheckEvent.java). If you can't find a reference to it in your fork, sounds like you messed up a rebase or something.

**Comment by LEOcab** [comment]
Okay I see it in the folder now... I might have been looking in the wrong folder because it wasn't there last night. Also github search still returns 0 results do I guess I just don't know how to search on here... legit said 0 results for this repository lol

This wasn't the file I was looking for either though... I don't see the checks. I need these checks, where are they now? https://pastebin.com/KPFtfhEE

**Reply by Jikoo** [reply]
Right inside Claim, same as before.

**Comment by LEOcab** [comment]
My custom code was right in between the "subdivision permission inheritance" check and the "failure message for all other cases" check.

**Reply by Jikoo** [reply]
Sounds like you don't need a custom fork any more because you can use a listener for the ClaimPermissionCheckEvent to change GP's behavior now.

**Reply by LEOcab** [reply]
Ouchie... Any chance you could show me how do use the new event? Would doing a .setCancel() on it be the exact equivalent of returning an empty string from allowBuild()?

**Reply by LEOcab** [reply]
In other words, how do I translate this: https://pastebin.com/HQmk7gz0 to the new system?

**Reply by Jikoo** [reply]
https://github.com/TechFortress/GriefPrevention/blob/6b87e036fa7c747c92be025e9266d98adf099b0c/src/main/java/me/ryanhamshire/GriefPrevention/events/ClaimPermissionCheckEvent.java#L17-L18
https://github.com/TechFortress/GriefPrevention/blob/6b87e036fa7c747c92be025e9266d98adf099b0c/src/main/java/me/ryanhamshire/GriefPrevention/events/ClaimPermissionCheckEvent.java#L126
https://github.com/TechFortress/GriefPrevention/blob/6b87e036fa7c747c92be025e9266d98adf099b0c/src/main/java/me/ryanhamshire/GriefPrevention/events/ClaimPermissionCheckEvent.java#L155-L163

Javadocs pretty much answer your question. It's preferred that you call `claimPermissionCheckEvent.setDenialReason(() -> "")` because it's much more clear what your code is doing, but that would have the same end result.

**Reply by LEOcab** [reply]
Ah I see! Alright after some fiddling I figured out how to convert the bulk of my code to the API.

Now I can't figure out how to access Claim.playerIDToClaimPermissionMap... it's still there but I can't access it from outside the plugin. Is there an API way of doing that? I just need to check the permission level of everyone in the claim

**Reply by Jikoo** [reply]
It's accessible, but not exactly how one might expect https://github.com/TechFortress/GriefPrevention/blob/6b87e036fa7c747c92be025e9266d98adf099b0c/src/main/java/me/ryanhamshire/GriefPrevention/GriefPrevention.java#L1462-L1468

**Reply by LEOcab** [reply]
That works! Thank you so much for all your help!! You were right, in the end this no longer has to be a fork :)
